### PR TITLE
Update of TestUpdateConnectionOnNSEChange

### PR DESCRIPTION
**Pull request by lazyniv**
Signed-off-by: Dmitry Vlasov <dmitry.vlasov@xored.com>

## Description
Fix nil pointer in updateConnection from nsm-monitor.go

## Motivation and Context
#2125

## How Has This Been Tested?

- [x] Covered by existing integration testing
- [ ] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: test/integration/recover_update_local_connection_test.go [hunk 1]
```diff
@@ -47,6 +47,10 @@ func TestUpdateConnectionOnNSEChange(t *testing.T) {
 
 	logrus.Infof("Waiting for NSMD to update NSC connection")
 	k8s.WaitLogsContains(nsc, "monitoring-nsc", "Connection updated", defaultTimeout)
+
+	logrus.Info("Waiting to verify that NSC was not restarted")
+	err = k8s.ShouldNotRestart(nsc, 5)
```
Comment (denis-tingajkin) [inline test/integration/recover_update_local_connection_test.go]: Let's pass `time.Second * 5` instead of `5`

File: test/kubetest/k8s.go [hunk 2]
```diff
@@ -718,6 +718,21 @@ func (k8s *K8s) DescribePod(pod *v1.Pod) {
 	}
 }
 
+//ShouldNotRestart checks that pod should not restart
+func (k8s *K8s) ShouldNotRestart(pod *v1.Pod, secondsForWait time.Duration) error {
```
Comment (denis-tingajkin) [inline test/kubetest/k8s.go]: We can assert inside this function. Please take a look at field `g` of structure `k8s`.

Comment (denis-tingajkin) [reply test/kubetest/k8s.go]:
```suggestion
func (k8s *K8s) ShouldNotRestart(pod *v1.Pod, duration time.Duration)  {
```

File: test/kubetest/k8s.go [hunk 3]
```diff
@@ -718,6 +718,21 @@ func (k8s *K8s) DescribePod(pod *v1.Pod) {
...
+	containerStatuses := pod.Status.ContainerStatuses
+	for start := time.Now(); time.Since(start) < time.Second*secondsForWait; {
```
Comment (denis-tingajkin) [inline test/kubetest/k8s.go]: This loop will load CPU. Let's add a wait interval.

File: test/kubetest/k8s.go [hunk 4]
```diff
@@ -718,6 +718,21 @@ func (k8s *K8s) DescribePod(pod *v1.Pod) {
...
+		for _, container := range containerStatuses {
```
Comment (denis-tingajkin) [inline test/kubetest/k8s.go]: Also, this loop will make a copy of `ContainerStatus` on each iteration. Let's use the indexing loop and pointers.

File: test/kubetest/k8s.go [hunk 5]
```diff
@@ -718,6 +718,21 @@ func (k8s *K8s) DescribePod(pod *v1.Pod) {
...
+			if container.State.Terminated != nil {
+				return errors.Errorf("The pod %v was restarted", pod.Name)
```
Comment (denis-tingajkin) [inline test/kubetest/k8s.go]: Error strings should not start with a capital letter. https://github.com/golang/go/wiki/Errors

Comment (denis-tingajkin) [reply test/kubetest/k8s.go]: Should we also put a reason from`state.Terminated` into error?

File: test/kubetest/k8s.go [hunk 6]
```diff
@@ -718,6 +718,21 @@ func (k8s *K8s) DescribePod(pod *v1.Pod) {
...
+			}
+		}
+		pod, _ = k8s.GetPod(pod)
```
Comment (denis-tingajkin) [inline test/kubetest/k8s.go]: Missed error checking.

File: test/integration/recover_update_local_connection_test.go [hunk 7]
```diff
@@ -47,6 +47,10 @@ func TestUpdateConnectionOnNSEChange(t *testing.T) {
 
 	logrus.Infof("Waiting for NSMD to update NSC connection")
 	k8s.WaitLogsContains(nsc, "monitoring-nsc", "Connection updated", defaultTimeout)
+
+	logrus.Info("Waiting to verify that NSC was not restarted")
```
Comment (denis-tingajkin) [inline test/integration/recover_update_local_connection_test.go]: Let's move this logging into a new assertation function and also add a name of the target pod.

File: side-cars/pkg/nsm-monitor/nsm-monitor.go [hunk 8]
```diff
@@ -197,7 +197,9 @@ func (c *nsmMonitorApp) readEvents(monitorClient connection.MonitorConnection_Mo
 func (c *nsmMonitorApp) updateConnection(conn *connection.Connection) {
 	if existingConn, exists := c.connections[conn.GetId()]; exists {
 		logrus.Infof(nsmMonitorLogWithParamFormat, "Connection updated", fmt.Sprintf("%v %v", existingConn, conn))
```
Comment (haiodo) [inline side-cars/pkg/nsm-monitor/nsm-monitor.go]: Please move message after all is updated.

File: test/integration/recover_update_local_connection_test.go [hunk 9]
```diff
@@ -47,6 +47,8 @@ func TestUpdateConnectionOnNSEChange(t *testing.T) {
 
 	logrus.Infof("Waiting for NSMD to update NSC connection")
 	k8s.WaitLogsContains(nsc, "monitoring-nsc", "Connection updated", defaultTimeout)
+
+	k8s.ShouldNotRestart(nsc, 5*time.Second, 1*time.Second)
```
Comment (haiodo) [inline test/integration/recover_update_local_connection_test.go]: Instead Please wait for message, since waiting for 5 seconds is not good.

**Pull request merged**